src/pages/registration/RegistrationFormTabs.tsx

```diff
@@ -29,17 +38,57 @@ const descriptionFieldNames = Object.values(DescriptionFieldNames);
 const referenceFieldNames = Object.values(ReferenceFieldNames);
 
 interface RegistrationFormTabsProps {
-  setTabNumber: (newTab: number) => void;
-  tabNumber: number;
+  setTabNumber: (newTab: RegistrationTab) => void;
+  tabNumber: RegistrationTab;
 }
 
 export const RegistrationFormTabs: FC<RegistrationFormTabsProps> = ({ setTabNumber, tabNumber }) => {
   const { t } = useTranslation('registration');
-  const { errors, touched, values } = useFormikContext<Registration>();
-  const {
-    entityDescription: { contributors },
-    fileSet: { files = [] },
-  } = values;
+  const { errors, touched, values, setTouched } = useFormikContext<Registration>();
+
+  const valuesRef = useRef(values);
+  useEffect(() => {
+    valuesRef.current = values;
+  }, [values]);
+
+  const touchedRef = useRef(touched);
+  useEffect(() => {
+    touchedRef.current = touched;
+  }, [touched]);
+
+  const highestPreviouslyTouchedTabRef = useRef<RegistrationTab | -1>(-1); // -1 is used to indicate that no tabs are touched yet
```

Comment:
hva med en konstant:  NoTabTouched =-1 isteden ?

Reply:
vi må vite hvilken tab som er den siste hittil besøkte, slik at vi også kan sette alle felt på tidligere tabs til touched. men vi trenger ikke å sette til touched for en tab som tidligere skal ha blitt satt til touched basert på hva som er hittil største tab (unngå dobbeltarbeid). ser ikke noen annen løsning på det... 🤔

Reply:
skjønner ikke helt hvorfor det heter `highest...` -  er ikke dette `previouslyTouchedTabRef`?

Reply:
mente forresten å bytte den kommentaren med en konstant for -1 med forklarende navn.

Reply:
Hvis man navigerer `tab 3 -> tab 1 -> tab 2` trenger vi ikke å gjøre noen on mount validering for `tab 1` og `tab 2`, siden de allerede har blitt validert når man var på `tab 3`. Vi trenger å vite hvilken tab lengst til høyre som er besøkt, for da vet vi at alle tabs til venstre for den skal valideres (hvis de ikke allerede er det, ref `highestPreviouslyTouchedTabRef `), @sarahse

Reply:
åjaa, skjønner 👍